pkg/condition/condition.go

```diff
@@ -111,3 +118,35 @@ func SetTerminal(
 	c.Reason = reason
 	subject.ReplaceConditions(allConds)
 }
+
+// SetLateInitialized sets the resource's Condition of type ConditionTypeLateInitialized to
+// the supplied status, optional message and reason.
+func SetLateInitialized(
+	subject acktypes.ConditionManager,
+	status corev1.ConditionStatus,
+	message *string,
+	reason *string,
+) {
+	allConds := subject.Conditions()
+	var c *ackv1alpha1.Condition
+	if c = LateInitialized(subject); c == nil {
+		c = &ackv1alpha1.Condition{
+			Type: ackv1alpha1.ConditionTypeLateInitialized,
+		}
+		allConds = append(allConds, c)
+	}
+	now := metav1.Now()
+	c.LastTransitionTime = &now
+	c.Status = status
+	c.Message = message
+	c.Reason = reason
+	subject.ReplaceConditions(allConds)
+}
```

Comment:
nit: I just saw that `SetTerminal` and `SetLateInitialized` are very similar, but ideally I think they should look like:

```go
func SetLateInitialized(
	subject acktypes.ConditionManager,
	status corev1.ConditionStatus,
	message *string,
	reason *string,
) {
	if c := LateInitialized(subject); c == nil {
		now := metav1.Now()
		c := &ackv1alpha1.Condition{
			Type:               ackv1alpha1.ConditionTypeLateInitialized,
			LastTransitionTime: &now,
			Status:             status,
			Reason:             reason,
		}
		allConds := subject.Conditions()
		allConds = append(allConds, c)
		subject.ReplaceConditions(allConds)
	}
}
```

Reply:
No, I do not think this will work the same. The code you provided does not update the LateInitialized condition, if it already exists.

Reply:
I agree there is some room for refactor to take out common code but it does not reduce code complexity by a lot.

Reply:
Ohhh yeah, gotcha! Totally makes sense!